src/string-class.js

```diff
@@ -0,0 +1,109 @@
+// Disable eslint do not modify native methods
+/* eslint-disable no-extend-native */
+
+String.prototype.hasVowels = function hasVowels() {
+  // Returns true if a vowel exists in the input string. Returns false otherwise.
+  const vowels = new RegExp('[aeiou]', 'i');
+  return vowels.test(this);
+};
+
+String.prototype.toUpper = function toUpper() {
+  /* Replace all lowercase letters in the input string with their uppercase
+  * analogs by converting each letter's ASCII value to decimal then back to
+  * ASCII
+  */
+  const upper = new RegExp('[a-z]', 'g');
+  return this.replace(upper, function transform(letter) {
+    return String.fromCharCode(letter.charCodeAt(0) - 32);
+  });
+};
+
+String.prototype.toLower = function toLower() {
+  const lower = new RegExp('[A-Z]', 'g');
+  return this.replace(lower, function transform(letter) {
+    return String.fromCharCode(letter.charCodeAt(0) + 32);
+  });
+};
+
+String.prototype.ucFirst = function ucFirst() {
+  /* Convert the first character of the input string to uppercase then concatenate
+  * the rest of the string with the uppercase character
+  */
+  return this.length ? this[0].toUpper() + this.slice(1) : '';
+};
+
+String.prototype.isQuestion = function isQuestion() {
+  // Returns true if the last letter of the input string a question mark. False otherwise
+  const questionable = new RegExp(/\?$/);
+  return questionable.test(this);
+};
+
+String.prototype.words = function words() {
+  /* Filter out any non word characters at the beginning and end of the input string.
+  * Split the new string at any non word character excluding the apostrophe.
+  * and return an array of words.
+  * i.e. Don't split at (A-Z,a-z,0-9 or ')
+  */
+  const parsedString = this.trim().replace(new RegExp(/^\W|\W$/g), '');
+  return parsedString.split(new RegExp(new RegExp(/[^A-Za-z0-9_']+/g)));
+};
+
+String.prototype.wordCount = function wordCount() {
+  return this.words().length;
+};
+
+String.prototype.toCurrency = function toCurrency() {
+  /* (/d) -> Matches digits from 0-9 and remembers the particular digit match
+  * (/d)(?=(\d{3})+\.) -> Lookahead - Only matches a digit if it is followed by
+  * three or other digits and if the entire string is a float.
+  * Upon encountering match, insert a comma character
+  */
+  const currency = new RegExp(/(\d)(?=(\d{3})+\.)/g);
```

Comment:
**Code Quality: Nitpick**
Use more descriptive variable names. Modified `curr` to `currency`
```
const currency = new RegExp(/(\d)(?=(\d{3})+\.)/g);
```